### ISSUE TITLE
Include the eth_getEncryptionPublicKey and eth_decrypt methods

### ISSUE THREAD

**Issue by dysbulic** [MetaMask/metamask-mobile#1807]
**Describe the usability problem**

Version 8 of the Metamask browser extension [added](https://github.com/MetaMask/metamask-extension/issues/1190) `eth_getEncryptionPublicKey` to return a Curve25519 encryption key derived from the account's private key. Also it added `eth_decrypt` to decrypt information encrypted with the aforementioned public key.

[This page](https://codepen.io/dysbulic/pen/zYqdPoz) demonstrates the functionality in brief.

**Expected behavior**

These functions are not available in the mobile version. It would be very useful for my application is they were.

------------------------------------------------------------------------------------------------------
_filled out after submission_

**User impact | Usage frequency | Feasibility scores**
_How important is this to the user, what the user wants to accomplish | Daily, weekly, monthly | Small, medium, large_

**Comment by mrenoon** [comment]
Hi, is there any timeline for this feature yet?

**Comment by rstormsf** [comment]
We need this feature for our mobile users.

**Comment by dysbulic** [comment]
Depending on the flexibility of your app, [IDX](//idx.xyz) has the ability to generate cyphertexts (JWEs) with keys associated with a $ETH account.

**Comment by omnat** [comment]
@mrenoon, no timeline yet. We'll update here when we plan for it. 
Curious to hear which dapp you are working on, and how you intend to use the `eth_getEncryptionPublicKey` & `eth_decrypt` methods

**Comment by rstormsf** [comment]
@omnat 
Currently, app.tornado.cash uses both methods for storing encrypted private key on-chain. Encryption happens with metamask's public key for recovery.

**Comment by boyuanx** [comment]
Our dApp also uses it to encrypt and decrypt AES keys that encrypt and decrypt various documents stored on IPFS.

**Comment by ArcQ** [comment]
is this in the roadmap? we would love this feature as well

**Comment by shawnmitchell** [comment]
Yes please! Parity with the browser extension with respect to this functionality would be most appreciated!

**Comment by joshuaohana** [comment]
Would also love this feature on mobile. I use this to enable users to keep private data onchain and decrypt only with their private keys - makes my app unusable on mobile without it...

**Comment by amilich** [comment]
Same!

**Comment by kwyszomierski-bluesoft** [comment]
Our app on web is using `eth_decrypt` to decipher files. Would be amazing to have it working on mobile as well.

**Comment by jdbertron** [comment]
Hello ?  Can we get some feedback for when this will be on the roadmap ?

**Comment by Sgoldik** [comment]
My marketplace dApp also uses these features, I would like to see them in the mobile version.

**Comment by davigp** [comment]
same here, we are limited to desktop experience 🙁 

curious about how we could help to implement `eth_getEncryptionPublicKey` for mobile

**Comment by prtk418** [comment]
Is this feature being worked on?

**Comment by avivash** [comment]
Folks from brave, metamask and various other CASA members are discussing a chain agnostic way to implement this for all wallet clients here:

https://github.com/ChainAgnostic/CAIPs/issues/132

and more actively here https://ethereum-magicians.org/t/eip-5630-encryption-and-decryption/10761/112

**Comment by mostafaamr137** [comment]
Hello ? Can we get some feedback for if this feature will ever be implemented ?

**Comment by Porianesu** [comment]
These two methods are still not supported now, is it possible to support them in the future.

**Comment by mark-smartseal** [comment]
Would really love it if they were. It would be incredibly useful
